From 8631e7afa40a752fe41cacc29a2a4c358a8a1746 Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Fri, 29 Jul 2022 17:50:56 +0100
Subject: [PATCH 1/2] gnu: Add posix-ipc

* gnu/packages/python-xyz.scm: (python-posix-ipc): New variable.
---
 gnu/packages/python-xyz.scm | 32 ++++++++++++++++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/gnu/packages/python-xyz.scm b/gnu/packages/python-xyz.scm
index ac7c0d6066..732cd2a376 100644
--- a/gnu/packages/python-xyz.scm
+++ b/gnu/packages/python-xyz.scm
@@ -29590,6 +29590,38 @@ (define-public python-queuelib
 fashion.")
     (license license:bsd-3)))
 
+(define-public python-posix-ipc
+  (package
+    (name "python-posix-ipc")
+    (version "1.0.5")
+    (source
+     (origin
+       ;; The source distributed on PyPI is prebuild.
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/osvenskan/posix_ipc")
+             (commit (string-append "rel" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "17y4d0pmvp199c5hbs602ailhlh9f9zv89kmpbd8jhyl6rgaxsvs"))))
+    (build-system python-build-system)
+    (arguments
+     (list #:phases
+           #~(modify-phases %standard-phases
+               (add-after 'unpack 'patch-cc-path
+                 (lambda _
+                   (substitute* "prober.py"
+                     (("cmd = .cc")
+                      (string-append "cmd = \"" #$(cc-for-target)))))))))
+    (native-inputs
+     (list python-unittest2))
+    (home-page "http://semanchuk.com/philip/posix_ipc/")
+    (synopsis "POSIX IPC primitives for Python")
+    (description
+     "This package provides POSIX IPC primitives - semaphores, shared memory and
+message queues for Python.")
+    (license license:bsd-3))) ; BSD like Copyright (c) 2018, Philip Semanchuk
+
 (define-public python-itemadapter
   (package
     (name "python-itemadapter")
-- 
2.36.1

